src/cascadia/TerminalApp/AppCommandlineArgs.cpp

```diff
@@ -0,0 +1,506 @@
+//   commandline to parse.
+std::vector<Commandline> AppCommandlineArgs::BuildCommands(const int argc, const wchar_t* argv[])
+{
+    std::vector<Commandline> commands;
+    commands.emplace_back(Commandline{});
+
+    // For each arg in argv:
+    // Check the string for a delimiter.
+    // * If there isn't a delimiter, add the arg to the current commandline.
+    // * If there is a delimiter, split the string at that delimiter. Add the
+    //   first part of the string to the current command, ansd start a new
+    //   command with the second bit.
+    for (auto i = 0; i < argc; i++)
+    {
+        _addCommandsForArg(commands, { argv[i] });
+    }
+
+    return commands;
+}
+
+// Function Description:
+// - Update and append Commandline objects for the given arg to the given list
+//   of commands. Each Commandline represents a single command to parse. These
+//   commands can be seperated by ";", which indicates the start of the next
+//   commandline. If the user would like to provide ';' in the text of the
+//   commandline, they can escape it as "\;".
+// - As we parse arg, if it doesn't contain a delimiter in it, we'll add it to
+//   the last command in commands. Otherwise, we'll generate a new Commandline
+//   object for each command in arg.
+// Arguments:
+// - commands: a list of Commandline objects to modify and append to
+// - arg: a single argument that should be parsed into args to append to the
+//   current command, or create more Commandlines
+// Return Value:
+// <none>
+void AppCommandlineArgs::_addCommandsForArg(std::vector<Commandline>& commands, std::wstring_view arg)
+{
+    std::wstring remaining{ arg };
+    std::wsmatch match;
+    // Keep looking for matches until we've found no unescaped delimiters,
+    // or we've hit the end of the string.
+    std::regex_search(remaining, match, AppCommandlineArgs::_commandDelimiterRegex);
+    do
+    {
+        if (match.size() == 0)
+        {
+            // Easy case: no delimiter. Add it to the current command.
+            commands.rbegin()->AddArg(remaining);
+            break;
+        }
+        else
+        {
+            // Harder case: There was a match.
+            const bool matchedFirstChar = match.position(0) == 0;
+            // If the match was at the beginning of the string, then the
+            // next arg should be "", since there was no content before the
+            // delimiter. Otherwise, add one, since the regex will include
+            // the last character of the string before the delimiter.
+            auto delimiterPosition = matchedFirstChar ? match.position(0) : match.position(0) + 1;
+            auto nextArg = remaining.substr(0, delimiterPosition);
+
+            if (nextArg != L"")
+            {
+                commands.rbegin()->AddArg(nextArg);
+            }
+
+            // Create a new commandline
+            commands.emplace_back(Commandline{});
+            commands.rbegin()->AddArg(std::wstring{ L"wt.exe" });

```

Comment:
should we use the real exe or just a sentinel

Reply:
We should not use hardcoded `wt.exe` as another resource, IMO. This should either be looking up its own name at runtime or using the sentinel as you say.